### ISSUE TITLE
[net] getaddrinfo(host='elasticsearch-logging-data.kubesphere-logging-system.svc', err=4): Domain name not found

### ISSUE THREAD

**Issue by zhangzhenguo100865** [kubesphere/kubesphere#4640]
外部es无法使用，fluent-bit 守护进程集错误
 [2022/01/22 11:53:01] [ warn] [net] getaddrinfo(host='elasticsearch-logging-data.kubesphere-logging-system.svc', err=4): Domain name not found

 [2022/01/22 11:53:01] [ warn] [engine] chunk '41-1642852363.984202683.flb' cannot be retried: task_id=0, input=tail.2 > output=es.0

 [2022/01/22 11:53:01] [ warn] [net] getaddrinfo(host='elasticsearch-logging-data.kubesphere-logging-system.svc', err=4): Domain name not found

 [2022/01/22 11:53:01] [ warn] [engine] chunk '41-1642852359.535730633.flb' cannot be retried: task_id=2, input=tail.2 > output=es.0

 [2022/01/22 11:53:04] [ warn] [net] getaddrinfo(host='elasticsearch-logging-data.kubesphere-logging-system.svc', err=4): Domain name not found

 [2022/01/22 11:53:04] [ warn] [engine] failed to flush chunk '41-1642852381.822283808.flb', retry in 9 seconds: task_id=0, input=tail.2 > output=es.0 (out_id=0)

 [2022/01/22 11:53:09] [ warn] [net] getaddrinfo(host='elasticsearch-logging-data.kubesphere-logging-system.svc', err=4): Domain name not found

 [2022/01/22 11:53:09] [ warn] [engine] failed to flush chunk '41-1642852384.842305242.flb', retry in 9 seconds: task_id=2, input=tail.2 > output=es.0 (out_id=0)

 [2022/01/22 11:53:11] [ warn] [net] getaddrinfo(host='elasticsearch-logging-data.kubesphere-logging-system.svc', err=4): Domain name not found

 [2022/01/22 11:53:11] [ warn] [engine] chunk '41-1642852369.161278598.flb' cannot be retried: task_id=1, input=tail.2 > output=es.0

 [2022/01/22 11:53:11] [ warn] [net] getaddrinfo(host='elasticsearch-logging-data.kubesphere-logging-system.svc', err=4): Domain name not found

 [2022/01/22 11:53:11] [ warn] [engine] chunk '41-1642852379.516467899.flb' cannot be retried: task_id=3, input=tail.2 > output=es.0

 [2022/01/22 11:53:13] [ warn] [net] getaddrinfo(host='elasticsearch-logging-data.kubesphere-logging-system.svc', err=4): Domain name not found

 [2022/01/22 11:53:13] [ warn] [engine] chunk '41-1642852381.822283808.flb' cannot be retried: task_id=0, input=tail.2 > output=es.0

 [2022/01/22 11:53:14] [ warn] [net] getaddrinfo(host='elasticsearch-logging-data.kubesphere-logging-system.svc', err=4): Domain name not found

 [2022/01/22 11:53:14] [ warn] [engine] failed to flush chunk '41-1642852393.587700684.flb', retry in 11 seconds: task_id=1, input=tail.2 > output=es.0 (out_id=0)

 [2022/01/22 11:53:14] [ warn] [net] getaddrinfo(host='elasticsearch-logging-data.kubesphere-logging-system.svc', err=4): Domain name not found

 [2022/01/22 11:53:14] [ warn] [engine] failed to flush chunk '41-1642852389.347648596.flb', retry in 11 seconds: task_id=0, input=tail.2 > output=es.0 (out_id=0)

 [2022/01/22 11:53:18] [ warn] [net] getaddrinfo(host='elasticsearch-logging-data.kubesphere-logging-system.svc', err=4): Domain name not found

 [2022/01/22 11:53:18] [ warn] [engine] chunk '41-1642852384.842305242.flb' cannot be retried: task_id=2, input=tail.2 > output=es.0

 [2022/01/22 11:53:19] [ warn] [net] getaddrinfo(host='elasticsearch-logging-data.kubesphere-logging-system.svc', err=4): Domain name not found

 [2022/01/22 11:53:19] [ warn] [engine] failed to flush chunk '41-1642852394.808018386.flb', retry in 8 seconds: task_id=2, input=tail.2 > output=es.0 (out_id=0)

 [2022/01/22 11:53:24] [ warn] [net] getaddrinfo(host='elasticsearch-logging-data.kubesphere-logging-system.svc', err=4): Domain name not found

 [2022/01/22 11:53:24] [ warn] [engine] failed to flush chunk '41-1642852399.364251521.flb', retry in 10 seconds: task_id=3, input=tail.2 > output=es.0 (out_id=0)

 [2022/01/22 11:53:26] [ warn] [net] getaddrinfo(host='elasticsearch-logging-data.kubesphere-logging-system.svc', err=4): Domain name not found

 [2022/01/22 11:53:26] [ warn] [engine] chunk '41-1642852393.587700684.flb' cannot be retried: task_id=1, input=tail.2 > output=es.0

 [2022/01/22 11:53:26] [ warn] [net] getaddrinfo(host='elasticsearch-logging-data.kubesphere-logging-system.svc', err=4): Domain name not found

 [2022/01/22 11:53:26] [ warn] [engine] chunk '41-1642852389.347648596.flb' cannot be retried: task_id=0, input=tail.2 > output=es.0

 [2022/01/22 11:53:27] [ warn] [net] getaddrinfo(host='elasticsearch-logging-data.kubesphere-logging-system.svc', err=4): Domain name not found

 [2022/01/22 11:53:27] [ warn] [engine] chunk '41-1642852394.808018386.flb' cannot be retried: task_id=2, input=tail.2 > output=es.0

 [2022/01/22 11:53:29] [ warn] [net] getaddrinfo(host='elasticsearch-logging-data.kubesphere-logging-system.svc', err=4): Domain name not found

 [2022/01/22 11:53:29] [ warn] [engine] failed to flush chunk '41-1642852404.496130375.flb', retry in 9 seconds: task_id=0, input=tail.2 > output=es.0 (out_id=0)

![image](https://user-images.githubusercontent.com/31802798/150637622-dd274076-607e-4f68-b54f-9bf797f6a5fb.png)

**Comment by wenchajun** [comment]
According to the error message, this is because the service does not exist. According to the error message, this is because the service does not exist. You can take a look at the official documentation.  https://kubesphere.io/docs/faq/observability/logging/
After doing so, you can try to restart the `ks-installer` pod.

**Comment by zhangzhenguo100865** [comment]
> 根据错误信息，这是因为该服务不存在。根据错误信息，这是因为该服务不存在。你可以看看官方文档。 https://kubesphere.io/docs/faq/observability/logging/ 这样做后，您可以尝试重新启动`ks-installer`pod。

I use 'KK' for installation. The relevant configurations are as follows. Would you please help me see if there is any error in this configuration?
![image](https://user-images.githubusercontent.com/31802798/150714618-e444110f-8dfe-402b-b891-1212942eacd3.png)

**Comment by zhangzhenguo100865** [comment]
> According to the error message, this is because the service does not exist. According to the error message, this is because the service does not exist. You can take a look at the official documentation. https://kubesphere.io/docs/faq/observability/logging/ After doing so, you can try to restart the `ks-installer` pod.

I use 'KK' for installation. The relevant configurations are as follows. Would you please help me see if there is any error in this configuration?
![image](https://user-images.githubusercontent.com/31802798/150715427-324f7333-1a4e-464d-bafb-18dbb8ec7bf2.png)

**Comment by wenchajun** [comment]
Thank you for the information, I think I understand why this error occurred. It was due to the fact that you changed the configuration and `fluentbit` didn't change it in time. You can restart `ks-installer`.Execute the following command
```
kubectl edit cc -n kubesphere-system ks-installer
```
Delete the red box and the information after it
![1642993857(1)](https://user-images.githubusercontent.com/39892300/150716092-21b89a12-33ef-4890-b79c-ebdbb52b664c.png)
then 
```
kubectl rollout restart deploy -n kubesphere-system ks-installer
```
Please ensure here that you are performing the actions in the documentation. https://kubesphere.io/docs/faq/observability/logging/

Incidentally, some of the fields in the information you provide can be coded， such as `externalElasticsearchHost`.

**Comment by zhangzhenguo100865** [comment]
> Thank you for the information, I think I understand why this error occurred. It was due to the fact that you changed the configuration and `fluentbit` didn't change it in time. You can restart `ks-installer`.Execute the following command
> 
> ```
> kubectl edit cc -n kubesphere-system ks-installer
> ```
> 
> Delete the red box and the information after it ![1642993857(1)](https://user-images.githubusercontent.com/39892300/150716092-21b89a12-33ef-4890-b79c-ebdbb52b664c.png) then
> 
> ```
> kubectl rollout restart deploy -n kubesphere-system ks-installer
> ```
> 
> Please ensure here that you are performing the actions in the documentation. https://kubesphere.io/docs/faq/observability/logging/
> 
> Incidentally, some of the fields in the information you provide can be coded， such as `externalElasticsearchHost`.

OK, thank you. I'll try.

**Comment by zhangzhenguo100865** [comment]
> Thank you for the information, I think I understand why this error occurred. It was due to the fact that you changed the configuration and `fluentbit` didn't change it in time. You can restart `ks-installer`.Execute the following command
> 
> ```
> kubectl edit cc -n kubesphere-system ks-installer
> ```
> 
> Delete the red box and the information after it ![1642993857(1)](https://user-images.githubusercontent.com/39892300/150716092-21b89a12-33ef-4890-b79c-ebdbb52b664c.png) then
> 
> ```
> kubectl rollout restart deploy -n kubesphere-system ks-installer
> ```
> 
> Please ensure here that you are performing the actions in the documentation. https://kubesphere.io/docs/faq/observability/logging/
> 
> Incidentally, some of the fields in the information you provide can be coded， such as `externalElasticsearchHost`.

After modifying according to the above information, the "fluent bit" error message has disappeared, but the log query interface still reports an error.
![image](https://user-images.githubusercontent.com/31802798/150721276-8fee229b-fcf0-4dcc-be08-103737f4f9d3.png)

**Comment by wenchajun** [comment]
Look at the `kubesphere` configmap, it's not the DNS service that's wrong here, it's the lack of the `http` service, check if you've changed the configuration wrong.
Did you follow the documentation?

**Comment by zhangzhenguo100865** [comment]
> Look at the `kubesphere` configmap, it's not the DNS service that's wrong here, it's the lack of the `http` service, check if you've changed the configuration wrong. Did you follow the documentation?

Thank you very much for your help. The problem has been solved. I modified "kubesphere config".
![image](https://user-images.githubusercontent.com/31802798/150725691-b1f6faf4-f63d-4145-86f4-05f9161c74bc.png)
I follow https://kubesphere.io/docs/pluggable-components/logging/ Configuration, but the document does not mention the need to modify kubesphere config.
